code-challenges.test.js

```diff
@@ -29,23 +59,129 @@ const colors2 = [
+// Test Suites: 1 failed, 1 total
+// Tests:       1 failed, 1 passed, 2 total
+// Snapshots:   0 total
+// Time:        0.185 s, estimated 1 s
+
 const votes1 = { upVotes: 13, downVotes: 2 }
 // Expected output: 11
 const votes2 = { upVotes: 2, downVotes: 33 }
 // Expected output: -31
 
 // b) Create the function that makes the test pass.
 
+// GREEN 
+// PSEUDOCODE 
+// input: an object that contains up votes and down votes
+// output: the end tally.
+
+// THE PROCESS:
+// create a function called tallyVotes that will take an object as an argument.
+// subtract the value of the downVotes property from the value of the upVotes property using math operators.
+// return the difference as the output of the function.
+
+const tallyVotes = (votes) => {
+  return votes.upVotes - votes.downVotes;
+}
+
 // --------------------3) Create a function that takes in two arrays as arguments and returns one array with no duplicate values. STRETCH: Use the spread operator to pass in a dynamic number of arguments.
 
 // a) Create a test with an expect statement using the variables provided.
 
+// RED 
+describe("mergeArrays", () => {
+  it("merges two arrays and removes duplicates", () => {
+    const dataArray1 = ["array", "object", "number", "string", "Boolean"]
+    const dataArray2 = ["string", "null", "Boolean", "string", "undefined"]
+    const mergedArray = mergeArrays(dataArray1, dataArray2)
+    expect(mergedArray).toHaveLength(7)
+    expect(mergedArray).toEqual(["array", "object", "number", "string", "Boolean", "null", "undefined"])
+  })
+})
+
+// ReferenceError: mergeArrays is not defined
+// Test Suites: 1 failed, 1 total
+// Tests:       1 failed, 2 passed, 3 total
+// Snapshots:   0 total
+// Time:        0.186 s, estimated 1 s
+
 const dataArray1 = ["array", "object", "number", "string", "Boolean"]
 const dataArray2 = ["string", "null", "Boolean", "string", "undefined"]
 // Expected output: ["array", "object", "number", "string", "Boolean", "null", "undefined"]
 
 // b) Create the function that makes the test pass.
+
+// GREEN 
+// PSEUDOCODE
+// input: two arrays as arguments
+// output: one array with no duplicate values
+
+// THE PROCESS:
+// create a function called mergeArrays that will take 2 arrays as arguments.
+// create a new variable called set that will use the Set built-in method to store the new values 
+// use a for loop to iterate over the full array(.length) starting at index 0(i = 0) moving through the array 1 index at a time(i++)
+// use .add to add new values to Set passing in 1 array index as an argument.
+// return the new array with the new values 
+
+
+const mergeArrays = (arr1, arr2) => {
+  const set = new Set()
+  for (let i = 0; i < arr1.length; i++) {
+    set.add(arr1[i])
+  }
+  for (let j = 0; j < arr2.length; j++) {
+    set.add(arr2[j])
+  }
+  return Array.from(set)
+}
+// REFACTOR 
+const mergedArrays = (arr1, arr2) => {
+  const set = new Set([...arr1, ...arr2])
+  return Array.from(set)
+}
```

Comment:
Love that you went and refactored your code to make it cleaner! Nice use of `Set` and `Array.from()`! Is there a way you could refactor it even more and turn it into a one-liner?